message.go

```diff
@@ -463,6 +463,7 @@ func (r *messageSetReaderV2) readMessage(min int64,
 				err = errShortRead
 				return
 			}
+			r.reader = bufio.NewReaderSize(r.reader, batchRemain)
```

Comment:
This could have unanticipated side effects for allocations...will need to think on this for a bit.

Reply:
The document mentions that "If the argument io.Reader is already a Reader with large enough size, it returns the underlying Reader." I think this shouldn't be an issue if the record sizes are within a range.

https://golang.org/pkg/bufio/#NewReaderSize

Reply:
Have a look at #250.  I believe it solves the same underlying issue, but it avoids an allocation and a copy in the case where the buffer needs to grow.  Sorry to be so slow...some of this code was recently contributed, so I'm not intimately familiar with it yet. 😅

Reply:
Thanks for your fix. I tested and it works. The turn around time is already very short.

Reply:
Great!  I'll merge it down then.